spec/unit/type/logical_volume_spec.rb

```diff
@@ -0,0 +1,337 @@
+        :size         => '10M',
+        :extents      => '1%vg',
+      )
+    }.to_not raise_error
+  end
+
+  it 'persistent which is not true or false raises error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name         => 'simba',
+        :ensure       => :present,
+        :volume_group => 'rafiki',
+        :size         => '10M',
+        :persistent   => 'nala',
+      )
+    }.to raise_error(Puppet::ResourceError,
+                     'Parameter persistent failed on Logical_volume[simba]: persistent must be either be true or false')
+  end
+
+  it 'persistent is true does not raise error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name         => 'simba',
+        :ensure       => :present,
+        :volume_group => 'rafiki',
+        :size         => '10M',
+        :persistent   => :true,
+      )
+    }.to_not raise_error
+  end
+
+  it 'minor not set to integer between 0 and 255 raises error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name         => 'ringo',
+        :ensure       => :present,
+        :volume_group => 'george',
+        :size         => '10M',
+        :minor        => 910
+      )
+    }.to raise_error(Puppet::ResourceError,
+                     'Parameter minor failed on Logical_volume[ringo]: 910 is not a valid value for minor. It must be an integer between 0 and 255')
+  end
+
+  it 'minor not set to integer between 0 and 255 raises error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name         => 'ringo',
+        :ensure       => :present,
+        :volume_group => 'george',
+        :size         => '10M',
+        :minor        => 1
+      )
+    }.to_not raise_error
+  end
+
+  it 'range set outside of valid range raises error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name         => 'john',
+        :ensure        => :present,
+        :volume_group  => 'paul',
+        :size          => '10M',
+        :range        => 'pete best'
+      )
+    }.to raise_error(Puppet::ResourceError,
+                     'Parameter range failed on Logical_volume[john]: pete best is not a valid range')
+  end
+
+  it 'range set within valid range does not raise error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name         => 'john',
+        :ensure        => :present,
+        :volume_group  => 'paul',
+        :size          => '10M',
+        # What is a valid range?
+        :range        => ['42', '0']
+      )
+    }.to_not raise_error
+  end
+
+  it 'invalid stripe count raises error' do
+    expect {
+      resource = Puppet::Type.type(:logical_volume).new(
+        :name         => 'bert',
+        :ensure        => :present,
+        :volume_group  => 'ernie',
+        :size          => '10M',
+        :stripes      => 12345
```

Comment:
I believe this is actually a valid number of stripse (given 12345 physical volumes) so shouldn't throw an error.